File: 5. Smallest Multiple/solution.ts
function findSmallestMultiple(maxDivisor: number): number {
    let smallestMultiple = 1;

    for (let i = 2; i <= maxDivisor; i++) {
        const gcd = calculateGCD(smallestMultiple, i);
        
        smallestMultiple = (smallestMultiple * i) / gcd;
    }

    return smallestMultiple;
}

function calculateGCD(a: number, b: number): number {
    while (b !== 0) {
        const temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

const result = findSmallestMultiple(20);
console.log(result);

File: 5. Smallest Multiple/solution.go
package main

import (
	"fmt"
)

func main() {
	result := findSmallestMultiple(20)
	fmt.Println(result)
}

func findSmallestMultiple(maxDivisor int) int {
	smallestMultiple := 1

	for i := 2; i <= maxDivisor; i++ {
		gcd := calculateGCD(smallestMultiple, i)

		smallestMultiple = (smallestMultiple * i) / gcd
	}

	return smallestMultiple
}

func calculateGCD(a, b int) int {
	for b != 0 {
		temp := b
		b = a % b
		a = temp
	}
	return a
}
